src/NaadSocketClient.php

```diff
@@ -155,9 +155,26 @@ protected function handleHeartbeat(SimpleXMLElement $xml)
             // Fetch, validate, then process missed alerts.
             foreach ( $missedAlerts as $alert ) {
                 $this->currentOutput = '';
-                $rawXml = $this->repositoryClient->fetchAlert(
-                    $alert['id'], $alert['sent']
-                );
+                try {
+                    $rawXml = $this->repositoryClient->fetchAlert(
+                        $alert['id'], $alert['sent']
+                    );
+                } catch (Exception $e) {
+                    // if the exception is a 500 error, log it and continue
+                    if ($e->getCode() === 500) {
```

Comment:
```suggestion
                    if ($e->getCode() >= 500) {
```
All 500+ level status codes should be handled this way.